python/mxnet/symbol/symbol.py

```diff
@@ -1475,13 +1482,19 @@ def optimize_for(self, backend, args=None, ctx=None, **kwargs):
         out = SymbolHandle()
         assert isinstance(backend, str)
 
-        if args is None:
+        if args is None or len(args) == 0:
             args = []
             args_handle = c_array(NDArrayHandle, [])
         else:
-            listed_arguments = self.list_arguments()
-            args_handle, args = self._get_ndarray_inputs('args', args, listed_arguments, False)
+            args_handle, args = self._get_ndarray_inputs('args', args,
+                                                         self.list_arguments(), False)
 
+        if aux is None or len(aux) == 0:
+            aux = []
```

Comment:
do we need this line?

Reply:
Yes, because later on line 1513 we do `mx_uint(len(aux))`